src/api/app/assets/javascripts/webui2/autocomplete.js

```diff
@@ -5,10 +5,19 @@ $(document).ready(function() {
       appendTo:  $(this).data('append'),
       source:    $(this).data('source'),
       minLength: 2,
+      create: function() {
+        // disable submit button
+        $('input.btn.btn-sm.btn-primary.px-4').prop("disabled", true);
+      },
       search: function() {
         $(this).prev().find('i').toggleClass('fa-search fa-spinner fa-spin');
       },
-      response: function() {
+      response: function(event, ui) {
```

Comment:
Guess this could move to the select event? Because this fires if I search for `Adm`. If I then click on Accept it tries to add the user `Adm`.

Reply:
I need both actually and change event. The select is for sure, the right event, but as soon as the user, delete a character or insert a new one, the button doesn't get disabled again. Using the change event. The response event as I implemented, should be able to enable/disable the button while you are typing.. not sure if it's an overkill or if depending how big is the list of possible entries, the interface will lag while iterating through the array with ```$.each```

Reply:
I think you're overdoing it. This exercise is about helping the user to input the right things, it is **not** about keeping them from putting in the wrong thing. We have validations for that.

Reply:
The issue is that we have a modal for this. We don't have any way to validate what the user submits and then opening up the modal again if somehow the user input was invalid. If this form was on its own page, this would be way simpler. Modals are bad for forms, we should have modals only for confirm with Yes/No.

Reply:
it's not about validating the user input and avoiding bad data, but usability.